strawberryfields/ops.py

```diff
@@ -757,13 +757,24 @@ class MeasureFock(Measurement):
 
     ns = None
 
-    def __init__(self, select=None):
+    def __init__(self, dark_counts=None, select=None):
```

Comment:
I guess this means the user either specifies the dark count rates for all the modes or for none of them, which I guess is fair.

Reply:
Yep, it wouldn't be difficult _per se_ to add the option to specify dark counts for some modes (or e.g. supply just one value that is applied to all modes) but I'm not sure the extra complexity is worth it.

Reply:
I suppose one use-case could be if the hardware at some point provides per-detector dark count info. Summoning @lneuhaus to see if this is likely 🙂 

If so, it should be relatively easy to do the following, right?

```python
MeasureFock(dark_counts=[0.1, 0.2]) | (q[0], q[1])
```

Reply:
> if the hardware at some point provides per-detector dark count info

We plan to provide information about coherent (=pump leakage) and incoherent (=Raman photons, blackbody radiation on the detector) "dark" count rates in the hardware certificate. This is relatively easy to measure. 

However, I don't see what the user can do about this by setting the dark count rate to a desired value (if we could control it on the hardware, we would always have it zero of course). What do you have in mind @josh146 ?

Reply:
Yes, this can be done. And I think it makes sense to be able to simulate hardware imperfections in this way. And passing the arguments in this way (e.g. rather than as a backend kwarg) makes sense as well, as this is really a detector property.

Reply:
Now it should be possible to add a dark-counts value to each measured mode (if `len(dark_counts)` is *not* equal to the number of measured modes, a `ValueError` is raised). Also, the samples are now always returned as a `np.ndarray`; before it depended on the number of shots.

Reply:
Awesome!